tools/src/main/scala/com/fulcrumgenomics/str/vcf/StrGenotypeDuplexMolecules.scala

```diff
@@ -0,0 +1,357 @@
+
+  Io.assertReadable(input)
+  Io.assertReadable(intervals)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+
+    def f(ext: String): FilePath = PathUtil.pathTo(output + ext)
+
+    val vcfIn           = new VCFFileReader(input.toFile, false)
+    val overlapDetector = loadIntervals()
+    val sampleName      = toSampleName(vcfIn.getFileHeader)
+    val vcfOut          = toVcfWriter(header=vcfIn.getFileHeader, sampleName=sampleName, output=f(".vcf.gz"))
+    val infoWriter      = Io.toWriter(f(".info.txt"))
+    val progress        = ProgressLogger(logger=logger, noun="variants")
+
+    vcfIn.foreach { ctx =>
+      val overlaps = overlapDetector.getOverlaps(ctx).toSeq
+      if (overlaps.isEmpty) {
+        logger.warning(s"Skipping variant call with no overlaps: $ctx")
+      }
+      else {
+        // Developer notes:
+        // - at a some point in the future, it would be nice to remove all of the INFO and FORMAT values we no longer
+        //   use
+
+        val strInfo = overlaps match {
+          case Seq()     => unreachable("Found an emtpy overlap")
+          case Seq(info) => info
+          case infos     =>
+            logger.warning(s"Found multiple overlaps for variant: $ctx")
+            infos.head
+        }
+
+        val ctxBuilder = new VariantContextBuilder(ctx)
+        val genotypes  = ctx.getGenotypes
+        val alleles    = ctx.getAlleles.toSeq
+
+        // get the count per allele
+        val counts: Seq[Int] =  alleles.map { allele => genotypes.map { genotype => genotype.countAllele(allele) }.sum }
+        require(alleles.length == counts.length, s"Found ${alleles.length} alleles and ${counts.length} counts")
+
+        // get the sample name, which should be the same for all "samples" (i.e. molecules) after stripping the trailing
+        // molecular id.
+        val sampleName: String = {
+          val sampleNames = genotypes.map(_.getSampleName)
+            .map { name =>
+              val index = name.lastIndexOf('-')
+              require(index > 0, s"Sample name '$name' does not have a trailing '-<mid>'")
+              name.substring(0, index)
+            }.toSet.toList
+          sampleNames match {
+            case Nil         => fail("No sample names found!")
+            case name :: Nil => name
+            case names       => fail(s"More than one sample name found: ${names.mkString(", ")}")
+          }
+        }
+
+        // get the new genotype
+        val refAlleleLength = ctx.getReference.length()
+        val allCalls = alleles.zip(counts)
+          .filter(_._2 > 0) // ignore zero counts
+          .map { case (allele, count) =>
+            val baseDiff     = allele.length() - refAlleleLength
+            val repeatLength = baseDiff + (strInfo.refLength * strInfo.unitLength)
+            StrAllele(allele=allele, repeatLength=repeatLength, count=count)
+          }
+          .sortBy(-_.count)
+
+        val genotypeCalls = {
```

Comment:
I think this needs to be marginally less naive.  I think you want something like:

1. Take the first allele
2. If the first allele doesn't represent > 80% (?) of the counts, also take the second allele

Otherwise anyone who is homozygous will get called het so long as there is at least one error.